samples/bluetooth/periodic_adv_rsp/src/main.c

```diff
@@ -303,7 +307,14 @@ int main(void)
 
 		printk("Scanning successfully started\n");
 
-		k_sem_take(&sem_connected, K_FOREVER);
+		/* Wait for either remote info available or involuntary disconnect */
+		k_poll(events, ARRAY_SIZE(events), K_FOREVER);
+		err = k_sem_take(&sem_connected, K_NO_WAIT);
+		if (err) {
+			printk("Disconnected before remote info available\n");
+
+			goto disconnect;
```

Comment:
You're right. I opted for one common label and a couple of redundant calls in the exception case. I'll change it to a separate label if that's preferred.

Reply:
Samples are often used as reference code by app developers to write their own apps. Not having a separate label might make someone think that in this case you actually have to do those things even though in practice they're unnecessary (there's not even a code comment pointing this out). Because of that I think it'd be better to have a separate label.

Reply:
Ok, no problem. I've now changed to a separate label in [0c7017e](https://github.com/zephyrproject-rtos/zephyr/pull/78954/commits/0c7017eca68b5fc9b59dc7e3cba8d98a5fb90b2e).